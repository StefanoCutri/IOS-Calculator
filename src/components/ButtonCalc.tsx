import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native';
import { styles } from './themes/appTheme';

interface Props {
    text: string,
    bgColor?: string,
    width?: boolean,
    onPress: (numberText: string) => void,
    testID?: string
}

export const ButtonCalc = ({text, bgColor = '#2D2D2D', width = false, onPress}: Props) => {

    return (
        
        <TouchableOpacity
            onPress={()=> onPress(text)}
            testID='btnCalc'
        >
        <View style={
           {
               ...styles.button,
               backgroundColor: bgColor,
               width: (width) ? 155 : 70
           }

        }>
            <Text style={{
                ...styles.buttonText,
                color: (bgColor === '#9B9B9B') ? 'black' : 'white'
                }}> {text} </Text>
        </View>

        </TouchableOpacity>

    )
}
