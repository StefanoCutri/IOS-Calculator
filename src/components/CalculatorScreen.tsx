import React from 'react'
import { Text, View } from 'react-native';
import { ButtonCalc } from './ButtonCalc';
import { styles } from './themes/appTheme';
import { useCalculator } from '../hooks/useCalculator';

export const CalculatorScreen = () => {

    const {
        result, 
        smallResult, 
        cleanResult,
        btnAdd,
        btnRest, 
        btnDelete, 
        btnDivide, 
        btnMultiply, 
        createNumber, 
        posistiveNegative, 
        calculate
        
    } = useCalculator();

    return (
        <View style={styles.container}>

            {
                (smallResult !== '0') 
                &&  <Text style={styles.smallResult}> {smallResult} </Text>
           
            }
            
            <Text 
            style={styles.result}
            numberOfLines={1}
            adjustsFontSizeToFit
            testID='result'
            >{result} 
            </Text>

            <View style={styles.row}>
                <ButtonCalc text='C' bgColor='#9B9B9B' onPress={cleanResult} testID='clean'/>
                <ButtonCalc text='+/-' bgColor='#9B9B9B' onPress={posistiveNegative}/>
                <ButtonCalc text='del' bgColor='#9B9B9B' onPress={btnDelete}/>
                <ButtonCalc text='/' bgColor='#FF9427' onPress={btnDivide}/>
            </View>

            <View style={styles.row}>
                <ButtonCalc text='7' onPress={createNumber}/>
                <ButtonCalc text='8' onPress={createNumber}/>
                <ButtonCalc text='9' onPress={createNumber}/>
                <ButtonCalc text='X' bgColor='#FF9427' onPress={btnMultiply}/>
            </View>
            
            <View style={styles.row}>
                <ButtonCalc text='4' onPress={createNumber}/>
                <ButtonCalc text='5' onPress={createNumber}/>
                <ButtonCalc text='6' onPress={createNumber}/>
                <ButtonCalc text='-' bgColor='#FF9427' onPress={btnRest}/>
            </View>
            
            <View style={styles.row}>
                <ButtonCalc text='1' onPress={createNumber}/>
                <ButtonCalc text='2' onPress={createNumber}/>
                <ButtonCalc text='3' onPress={createNumber}/>
                <ButtonCalc text='+' bgColor='#FF9427' onPress={btnAdd}/>
            </View>
            
            <View style={styles.row}>
                <ButtonCalc text='0' onPress={createNumber}width/>
                <ButtonCalc text='.' onPress={createNumber}/>
                <ButtonCalc text='=' bgColor='#FF9427' onPress={calculate}/>
            </View>
            

        </View>

    )
}
