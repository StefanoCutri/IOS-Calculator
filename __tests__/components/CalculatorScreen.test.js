import React from 'react';

import {render, fireEvent } from '@testing-library/react-native'
import { act } from 'react-test-renderer';

import { CalculatorScreen } from '../../src/components/CalculatorScreen'

let wrapper = render(<CalculatorScreen/>);
let btnCalc = wrapper.getAllByTestId('btnCalc');

describe('<CalculatorScreen/>', () => {
    
    beforeEach(()=>{
        wrapper = render(<CalculatorScreen/>);
        btnCalc = wrapper.getAllByTestId('btnCalc');
    })

    it('renders correctyly', ()=> {

        expect(wrapper).toMatchSnapshot();
    })

    it('should change the result and clean it', () => {

        const result = wrapper.getByTestId('result')

        act(()=> {

            //press button "7"
            fireEvent(btnCalc[4], 'onPress')

        })
        act(()=> {

            //press button "7"
            fireEvent(btnCalc[4], 'onPress')

            //now the number is 77

        })

        expect(result.props.children).toBe('77')

        act(()=> {

            //Press button "C"
            fireEvent(btnCalc[0], 'onPress')

        })

        expect(result.props.children).toBe('0')

    })

    it('change the num to negative', ()=> {

        const result = wrapper.getByTestId('result')

        act(()=> {

            //press button "+/1"
            fireEvent(btnCalc[1], 'onPress')

        })

        expect(result.props.children).toBe('-0')

    
    })

    it('should create the number', ()=> {

        const result = wrapper.getByTestId('result')

        act(()=> {

            //press button "7"
            fireEvent(btnCalc[4], 'onPress')
     
        })
        act(()=> {

            //press button "8"
            fireEvent(btnCalc[5], 'onPress')
        })
        act(()=> {
         
            //press button "9"
            fireEvent(btnCalc[6], 'onPress')
        })

        expect(result.props.children).toBe('789')

    });

    it('should delete the last number and be 0 if there is no more', ()=> {

        const result = wrapper.getByTestId('result')

        act(()=> {

            //press button "7"
            fireEvent(btnCalc[4], 'onPress')
     
        })
        act(()=> {

            //press button "8"
            fireEvent(btnCalc[5], 'onPress')

            // now the number is 78
        })
        expect(result.props.children).toBe('78')
        
        act(()=> {

            //press button "del"
            fireEvent(btnCalc[2], 'onPress')

            // now the number should be 7
        })
        expect(result.props.children).toBe('7')
        
    })

    it('should not add to dots', ()=> {

        const result = wrapper.getByTestId('result')

        act(()=> {

            //press button "."
            fireEvent(btnCalc[17], 'onPress')
     
        })

        act(()=> {

            //press button "."
            fireEvent(btnCalc[17], 'onPress')
     
        })

        act(()=> {

            //press button "."
            fireEvent(btnCalc[17], 'onPress')
     
        })

        expect(result.props.children).toBe('0.')

    })
    

})
