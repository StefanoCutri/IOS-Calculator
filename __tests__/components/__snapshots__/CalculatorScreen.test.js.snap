// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorScreen/> renders correctyly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-end",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      Object {
        "color": "white",
        "fontSize": 60,
        "marginBottom": 10,
        "textAlign": "right",
      }
    }
    testID="result"
  >
    0
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#9B9B9B",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          C
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#9B9B9B",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          +/-
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#9B9B9B",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          del
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FF9427",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          /
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          7
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          8
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          9
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FF9427",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          X
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          4
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          5
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          6
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FF9427",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          -
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          1
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          2
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          3
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FF9427",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          +
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 155,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          0
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2D2D2D",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          .
           
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btnCalc"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FF9427",
            "borderRadius": 100,
            "height": 70,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "width": 70,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          =
           
        </Text>
      </View>
    </View>
  </View>
</View>
`;
